src/transactions/feed/TransactionDetailsScreen.tsx

```diff
@@ -104,23 +118,109 @@ function TransactionDetailsScreen({ navigation, route }: Props) {
       break
     case TokenTransactionTypeV2.SwapTransaction:
       content = <SwapContent exchange={transaction as TokenExchange} />
+      retryHandler = () => navigate(Screens.SwapScreenWithBack)
       break
   }
 
+  const transactionNetwork = networkIdToNetwork[transaction.networkId]
+
+  const openBlockExplorerHandler =
+    transaction.networkId in NetworkId
+      ? () =>
+          navigate(Screens.WebViewScreen, {
+            uri: new URL(
+              transaction.transactionHash,
+              blockExplorerUrls[transaction.networkId].baseTxUrl
+            ).toString(),
+          })
+      : undefined
+
+  const primaryActionHanlder =
+    transaction.status === TransactionStatus.Failed ? retryHandler : openBlockExplorerHandler
+
   return (
-    <ScrollView contentContainerStyle={styles.contentContainer}>
-      <SafeAreaView style={styles.content}>{content}</SafeAreaView>
+    <ScrollView contentContainerStyle={styles.container}>
+      <SafeAreaView edges={['right', 'bottom', 'left']}>
```

Comment:
I'm not entirely sure `left` and `right` are needed. Probably already handled by the wrapper navigator.